### PR TITLE
Added an user friendly error message for default phone number errors …

### PR DESCRIPTION
**Pull request by mujafar7**
…from fire base
Image for the Issue

Image after adding user friendly error message
Screenshot_20231120_150740
Screenshot_20231120_114010

### CODE REVIEW

File: src/pages/Login.tsx [hunk 1]
```diff
@@ -347,7 +358,8 @@ const Login: React.FC = () => {
     } catch (error) {
       console.log("Resend Otp Failed With Error " + error);
       setSentOtpLoading(false);
-      alert("Resend Otp Failed " + error);
+      //When Resend OTP Failed
+      alert(t("Resend Otp Failed!! You have entered the OTP incorrectly many times. Please try again after some time."));
```
Comment (Sutara-Skanda) [inline src/pages/Login.tsx:362]: put the all messages in constants page

File: src/pages/Login.tsx [hunk 2]
```diff
@@ -242,21 +242,32 @@ const Login: React.FC = () => {
...
+            if (error.includes("blocked all requests")) {
+                errorMessage = "Hi, Something went wrong. Please try again later in sometime.";
+            } else if (error.includes("E.164 format")) {
+                errorMessage = "Sorry, this is the incorrect phone number format. Please check and re-enter again";
+            }
+    
+            alert(t(errorMessage));
+        } else {
+            // Default error message for non-string errors
+            alert("Phone Number signin Failed. Something went wrong. Please try again later.");
```
Comment (Sutara-Skanda) [inline src/pages/Login.tsx:263]: translate the error message

File: src/pages/Login.tsx [hunk 3]
```diff
@@ -242,21 +242,32 @@ const Login: React.FC = () => {
...
-      // //@ts-ignore
-      // window.recaptchaVerifier.clear();
+    }catch (error) {
+        console.log("Phone Number signin Failed ");
+        setSpinnerLoading(false);
+        setSentOtpLoading(false);
+    
+        if (typeof error === 'string') {
+            // Handle the error as a string
+            let errorMessage = "Phone Number signin Failed. Something went wrong. Please try again later.";
```
Comment (Sutara-Skanda) [inline src/pages/Login.tsx:252]: call alert in common way

File: src/pages/Login.tsx [hunk 4]
```diff
@@ -242,21 +242,32 @@ const Login: React.FC = () => {
...
-      // //@ts-ignore
-      recaptchaVerifier!.clear();
-      // //@ts-ignore
-      // window.recaptchaVerifier.clear();
+    }catch (error) {
+        console.log("Phone Number signin Failed ");
+        setSpinnerLoading(false);
+        setSentOtpLoading(false);
+    
+        if (typeof error === 'string') {
```
Comment (Sutara-Skanda) [inline src/pages/Login.tsx:250]: if possible get error code from error type and show the error message according to error code

**Review by Sutara-Skanda — CHANGES_REQUESTED** [review]

**Review by Sutara-Skanda — APPROVED** [review]

**Comment by mujafar7** [comment]
Screenshot_20231128_094448
Screenshot_20231128_102326

**Review by Sutara-Skanda — APPROVED** [review]

**Pull request merged**